src/lib/y2partitioner/widgets/configure.rb

```diff
@@ -0,0 +1,280 @@
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "cwm"
+require "y2partitioner/widgets/reprobe"
+
+Yast.import "Stage"
+Yast.import "Popup"
+Yast.import "Arch"
+Yast.import "PackageCallbacks"
+Yast.import "PackageSystem"
+
+module Y2Partitioner
+  module Widgets
+    # "Configure" menu button used to run the corresponding YaST clients to
+    # activate several storage technologies
+    #
+    # Most of the behavior of this widget is a direct translation of the old
+    # Yast::PartitioningEpAllInclude from yast-storage
+    class Configure < CWM::CustomWidget
+      include Reprobe
+
+      # Constructor
+      def initialize
+        textdomain "storage"
+        super
+      end
+
+      # Content of the widget, a menu button with the list of available
+      # configuration clients or an empty widget if no client is available
+      #
+      # @macro seeCustomWidget
+      #
+      # @return [Yast::WidgetTerm]
+      def contents
+        @contents ||= actions.empty? ? Empty() : MenuButton(Opt(*opt), label, items)
+      end
+
+      # Event handler for the configuration menu
+      #
+      # @param event [Hash] UI event
+      # @return [:reprobe, nil] :reprobe when some configuration client was
+      #   executed; nil otherwise.
+      def handle(event)
+        action = Action.find(event["ID"])
+
+        return nil unless action
+        return nil unless warning_accepted?(action) && availability_ensured?(action)
+
+        Yast::WFM.call(action.client)
+        reprobe
+        :reprobe
+      end
+
+    private
+
+      # @return [Array<Yast::WidgetTerm>]
+      def items
+        actions.map do |action|
+          Item(Id(action.id), Yast::Term.new(:icon, action.icon), action.label)
+        end
+      end
+
+      # @return [Array<Symbol>]
+      def opt
+        [:key_F7]
+      end
+
+      # Localized label for the menu button
+      #
+      # @return [String]
+      def label
+        # Translators: Configure menu in the initial Partitioner screen
+        _("Configure...")
+      end
+
+      # @macro seeCustomWidget
+      #
+      # Redefined in this class because the base implementation at CWM::CustomWidget
+      # does not search for ids into the items of a MenuButton.
+      def ids_in_contents
+        actions.map(&:id)
+      end
+
+      # Available actions to be offered to the user
+      #
+      # @return [Array<Action>]
+      def actions
+        @actions ||=
+          if Yast::Stage.initial
```

Comment:
Why to use `Yast::Stage.initial` instead of `Yast::Mode.instalation` ?

Reply:
Because is what the old code for the same button in yast2-storage did. And same applies about the decision whether to re-activate or not, in the past `Stage` was used for it.

Reply:
And there are other good reasons. In an AutoYaST installation, the second stage runs already in the target system, after the first reboot (but I assume `Mode.installation` is still true). In that situation we want the Partitioner (and Y2Storage in general) to behave like if it's working in an already installed system.